src/data_provider/src/osinfo/sysOsParsers.cpp

```diff
@@ -392,6 +391,20 @@ bool MacOsParser::parseSwVersion(const std::string& in, nlohmann::json& output)
     return ret;
 }
 
+bool MacOsParser::parseSystemProfiler(const std::string& in, nlohmann::json& output)
+{
+    constexpr auto SEPARATOR{':'};
+    static const std::map<std::string, std::string> KEY_MAP
+    {
+        {"System Version", "os_name"},
+    };
+    std::stringstream data{in};
+    const auto ret{ parseUnixFile(KEY_MAP, SEPARATOR, data, output) };
+    output["os_name"] = Utils::split(output["os_name"].get_ref<const std::string&>(), ' ').at(0);
```

Comment:
`ret` should be checked before splitting with spaces. Maybe using and validating that System Version against a regex is safer